datahub-web-react/src/app/entityV2/document/summary/RelatedSection.tsx

```diff
@@ -0,0 +1,232 @@
+import { Button } from '@components';
+import React, { useMemo } from 'react';
+import styled from 'styled-components';
+
+import { useUserContext } from '@app/context/useUserContext';
+import { AddRelatedEntityDropdown } from '@app/entityV2/document/summary/AddRelatedEntityDropdown';
+import { SectionContainer } from '@app/entityV2/shared/summary/HeaderComponents';
+import { EntityLink } from '@app/homeV2/reference/sections/EntityLink';
+import { useEntityRegistry } from '@app/useEntityRegistry';
+import colors from '@src/alchemy-components/theme/foundations/colors';
+
+import { AndFilterInput, DocumentRelatedAsset, DocumentRelatedDocument, EntityType, FilterOperator } from '@types';
+
+const SectionHeader = styled.div`
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    margin-bottom: 8px;
+`;
+
+const SectionTitle = styled.h4`
+    font-size: 16px;
+    font-weight: 600;
+    margin: 0;
+    color: ${colors.gray[600]};
+`;
+
+const List = styled.div`
+    display: flex;
+    flex-direction: column;
+    gap: 4px;
+`;
+
+const EmptyState = styled.div`
+    font-size: 14px;
+    font-weight: 400;
+    color: ${colors.gray[1800]};
+    text-align: center;
+    padding: 8px;
+`;
+
+const EntityItemContainer = styled.div`
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    gap: 8px;
+    width: 100%;
+
+    /* Show trash button only on hover */
+    .trash-button {
+        opacity: 0;
+        transition: opacity 0.2s ease;
+    }
+
+    &:hover .trash-button {
+        opacity: 1;
+    }
+`;
+
+const EntityLinkWrapper = styled.div`
+    flex: 1;
+    min-width: 0; /* Allow flex item to shrink below content size */
+`;
+
+const TrashButton = styled(Button)`
+    flex-shrink: 0;
+    padding: 0;
+`;
+
+interface RelatedSectionProps {
+    relatedAssets?: DocumentRelatedAsset[];
+    relatedDocuments?: DocumentRelatedDocument[];
+    documentUrn: string;
+    onAddEntities: (assetUrns: string[], documentUrns: string[]) => Promise<void>;
+    onRemoveEntity?: (urn: string) => Promise<void>;
+    canEdit?: boolean;
+}
+
+export const RelatedSection: React.FC<RelatedSectionProps> = ({
+    relatedAssets,
+    relatedDocuments,
+    documentUrn,
+    onAddEntities,
+    onRemoveEntity,
+    canEdit = true,
+}) => {
+    const entityRegistry = useEntityRegistry();
+    const userContext = useUserContext();
+
+    // Entity types that can be related (based on RelatedAsset.pdl)
+    // I don't love this, but I do want to enable searching for documents
```

Comment:
yeah i feel that we will definitely forget to add something to this list in the future..

that being said i don't see an obvious solution without muddying up our API for a specific use case like this

Reply:
Agree..